doc/verification.rst

```diff
@@ -5,27 +5,63 @@ Overview
+<https://github.com/google/riscv-dv>`_ to generate compiled instruction binaries, loads them into a
+simple memory mode, stimulates the Ibex core to run this program in memory, and then compares the
+core trace log against a golden model ISS trace log to check for correctness of execution.
 
-Testbench component
-~~~~~~~~~~~~~~~~~~~
+Testbench Architecture
+----------------------
 
--  Random instruction generator : `RISCV-DV <https://github.com/google/riscv-dv>`_
--  `Memory interface agent for instruction fetch and load/store
-   operations <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/ibex_mem_intf_agent>`_
--  `Interrupt interface agent <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/irq_agent>`_
--  `Memory model <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/mem_model>`_
--  `Test and sequence library <https://github.com/lowRISC/ibex/tree/master/dv/uvm/tests>`_
+As previously mentioned, this testbench has been constructed based on its usage of the RISCV-DV
+random instruction generator developed by Google. A block diagram of the testbench is below.
+
+.. figure:: images/tb.svg
+
+Memory Interface Agents
+~~~~~~~~~~~~~~~~~~~~~~
+
+Code can be seen `here
+<https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/ibex_mem_intf_agent>`_.
+Two of these agents are instantiated within the testbench, one for the instruction fetch interface,
+and the second for the LSU interface. These agents run slave sequences that wait for memory requests from
+the core, and then grant the requests for instructions or for data.
+
+Interrupt Interface Agent
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Code can be seen `here <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/irq_agent>`_.
+This agent is used to drive stimulus onto the Ibex core's interrupt pins randomly during test
+execution.
+
+Memory Model
+~~~~~~~~~~~~
+
+Code can be seen `here <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/mem_mode>`_.
+The testbench instantiates a single instance of this memory model that it loads the compiled
+assembly test program into at the beginning of each test. This acts as a unified instruction/data
+memory that serves all requests from both of the memory interface agents.
+
+Test and Sequence Library
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Code can be seen `here <https://github.com/lowRISC/ibex/tree/master/dv/uvm/tests>`_.
+The tests here are the main sources of external stimulus generation and checking for this testbench,
+as the memory interface slave sequences simply serve the core's memory requests.
+The tests here are all extended from core_ibex_base_test, and coordinate the entire flow for a
+single test, from loading the compiled assembly binary program into the testbench memory model, to
+checking the Ibex core status during the test and dealing with test timeouts.
+The sequences here are used to drive interrupt and debug stimulus into the core.
```

Comment:
I think it's better to add some descriptions on the checking mechanism, including ISS co-sim trace compare, and handshake mechanism.

Reply:
Agreed, my plan is to write up the detailed documentation of the handshaking in the riscv-dv repository, and then link it in this Ibex doc, and briefly explain how it's used. I'll add this in the verification.rst documentation as a separate section after I write up the handshake document for riscv-dv.